Monika-After-Story/MonikaModDev
# Adding a new Custom Graphics Menu

#1275 
Because the built-in menu causes us some problems when activated, I've created a custom renderer selection menu that looks nice and doesn't break flow.

Main thing is controlled by a new custom screen. Looks like this:
![holysexymenu](https://user-images.githubusercontent.com/3499462/38965857-bd7d3500-4344-11e8-8590-d6d9a0a5d8b1.png)


Renderer selection can be forced by an `environment.txt` file containing the environment var `RENPY_RENDERER`. Setting that to one of the available renderers will force renpy to run using that renderer.

Testing:
1. Open the settings screen.
2. Click `Change Renderer` under `Graphics` label.
3. Select a renderer.
4. Click yes at the prompt.
5. Click OK
6. Game will shut down.
7. **Open DDLC.exe/.sh** Running the game via Renpy SDK will **not** read the env file.
8. Reopen the Renderer menu. The renderer you selected should be marked as Current Renderer. The corresponding button should be disabled.

You can also check the value of `renpy.config.renderer`. (`auto`, `gl`, `angle`, `sw`)
You can also check the value of the `environment.txt` file. (For `auto`, the file should be deleted. If not, the file will have `auto` as the renderer)

### Additonal:
1. I've changed the default trigger for a `MASButtonDisplayable` to be MouseUP, which makes more sense since the other built-in screens use MouseUp as well.
2. Non-windows devices should **not** have the Angle/DirectX button available.

### NOTE:
* Software rendering sucks. Don't use it.
* This pull request has no dialogue. Therefore, only 1 review is necessary.
